server/src/main/java/org/elasticsearch/action/bulk/BulkShardRequest.java

```diff
@@ -44,6 +48,30 @@ public BulkShardRequest(ShardId shardId, RefreshPolicy refreshPolicy, BulkItemRe
         setRefreshPolicy(refreshPolicy);
     }
 
+    /**
+     * Public for test
+     * Set the transient metadata indicating that this request requires running inference before proceeding.
+     */
+    public void setInferenceFieldMap(Map<String, InferenceFieldMetadata> fieldInferenceMap) {
+        this.inferenceFieldMap = fieldInferenceMap;
+    }
+
+    /**
+     * Consumes the inference metadata to execute inference on the bulk items just once.
+     */
+    public Map<String, InferenceFieldMetadata> consumeInferenceFieldMap() {
+        Map<String, InferenceFieldMetadata> ret = inferenceFieldMap;
+        inferenceFieldMap = null;
+        return ret;
+    }
```

Comment:
Good catch - adding them to serialization.

> Could you add a failure in the wire serialization if inferenceFieldMap is null?

It could be null in case the inference information has already been consumed - the idea is avoiding to perform inference multiple times in case it has already been done. When consuming inference information, it is set to null to signal it.

I'm changing the `consumeInferenceFieldMap` to set it to an empty map to ensure we always have a value.

Reply:
Changed in 4240f6b38c0c56e5d90505af01f299e3b3277438